x-pack/legacy/plugins/maps/public/layers/vector_layer.js

```diff
@@ -376,15 +377,21 @@ export class VectorLayer extends AbstractLayer {
       const leftSourceName = await this.getSourceName();
       const {
         propertiesMap
-      } = await joinSource.getPropertiesMap(searchFilters, leftSourceName, join.getLeftFieldName());
+      } = await joinSource.getPropertiesMap(
+        searchFilters,
+        leftSourceName,
+        join.getLeftFieldName(),
+        registerCancelCallback.bind(null, requestToken));
       stopLoading(sourceDataId, requestToken, propertiesMap);
       return {
         dataHasChanged: true,
         join: join,
         propertiesMap: propertiesMap,
       };
     } catch (e) {
-      onLoadError(sourceDataId, requestToken, `Join error: ${e.message}`);
+      if (!(e instanceof DataRequestAbortError)) {
```

Comment:
Maybe this is what you were after by making it a separate Error class

Reply:
> Maybe this is what you were after by making it a separate Error class

exactly.

VectorLayer just needs to look for a generic DataRequestAbortError and does not need to know source implementation details like how SearchSource throws exceptions when aborted